docs/weave-tutorial/07-handlers.md

```diff
@@ -0,0 +1,675 @@
+func (h CreateArticleHandler) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx) (*weave.CheckResult, error) {
+    msg, _, err := h.validate(ctx, store, tx)
+    if err != nil {
+        return nil, err
+    }
+
+    // Calculate gas cost
+    gasCost := int64(len(msg.Content)) * newArticleCost / articleCostUnit
+
+    return &weave.CheckResult{GasAllocated: gasCost}, nil
+}
+```
+
+In the case of an Article creation, we decided to charge the author 1 gas per mile characters with the first 1000 characters offered: `gasCost := int64(len(msg.Content)) * newArticleCost / articleCostUnit`
+
+## Testing Handlers
+
+In order to test a handler, we need four things :
+
+- A storage
+- A weave context
+- An Authenticator associated with our context
+- A Tx object to process (eg. to check or to deliver)
+
+There is a ready to use in memory storage available in the [store package](https://github.com/iov-one/weave/blob/master/store/btree.go#L31-L36). There are also util functions available that we can use to create a weave context with a list of signers (eg. authorized addresses) via an [Authenticator](weave/design/permissions)
+The function below shows how to use them:
+
+```go
+func TestCreateArticle(t *testing.T) {
+    blogOwner := weavetest.NewCondition()
+    signer := weavetest.NewCondition()
+
+    now := weave.AsUnixTime(time.Now())
+    past := now.Add(-1 * 5 * time.Hour)
+    future := now.Add(time.Hour)
+
+    ownedBlog := &Blog{
+        Metadata:    &weave.Metadata{Schema: 1},
+        ID:          weavetest.SequenceID(1),
+        Owner:       signer.Address(),
+        Title:       "Best hacker's blog",
+        Description: "Best description ever",
+        CreatedAt:   past,
+    }
+    notOwnedBlog := &Blog{
+        Metadata:    &weave.Metadata{Schema: 1},
+        ID:          weavetest.SequenceID(2),
+        Owner:       blogOwner.Address(),
+        Title:       "Worst hacker's blog",
+        Description: "Worst description ever",
+        CreatedAt:   past,
+    }
+
+    cases := map[string]struct {
+        msg             weave.Msg
+        signer          weave.Condition
+        expected        *Article
+        wantCheckErrs   map[string]*errors.Error
+        wantDeliverErrs map[string]*errors.Error
+    }{
+        "success": {
+            msg: &CreateArticleMsg{
+                Metadata: &weave.Metadata{Schema: 1},
+                BlogID:   ownedBlog.ID,
+                Title:    "insanely good title",
+                Content:  "best content in the existence",
+                DeleteAt: future,
+            },
+            signer: signer,
+            expected: &Article{
+                Metadata:     &weave.Metadata{Schema: 1},
+                ID:           weavetest.SequenceID(1),
+                BlogID:       ownedBlog.ID,
+                Owner:        signer.Address(),
+                Title:        "insanely good title",
+                Content:      "best content in the existence",
```

Comment:
`                Content:  "best content in existence",
`

Reply:
These are golang code examples so there is no need to change them.